WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt

```diff
@@ -299,20 +302,52 @@ class OrderCreateEditViewModel @Inject constructor(
     private fun Order.hasProducts() = items.any { it.quantity > 0 }
 
     fun startScan() {
-        launch {
+        viewModelScope.launch {
             codeScanner.startScan().collect { status ->
                 when (status) {
                     is CodeScannerStatus.Failure -> {
                         // TODO handle failure case
                     }
                     is CodeScannerStatus.Success -> {
-                        // TODO handle success case
+                        viewState = viewState.copy(isUpdatingOrderDraft = true)
+                        fetchProductBySKU(status.code)
                     }
                 }
             }
         }
     }
 
+    private fun fetchProductBySKU(sku: String) {
+        val selectedItems = orderDraft.value?.items?.map { item ->
+            if (item.isVariation) {
+                SelectedItem.ProductVariation(item.productId, item.variationId)
+            } else {
+                Product(item.productId)
```

Comment:
```suggestion
                SelectedItem.Product(item.productId)
```
Nit: We could add the base class prefix to match the line above.

Reply:
Done https://github.com/woocommerce/woocommerce-android/pull/9053/commits/9504856c723e1faa107cd0678ce900844dc3cf31